custom_components/babybuddy/sensor.py

```diff
@@ -1,409 +1,359 @@
-            ATTR_CHILD: hass.states.get(call.data.get(ATTR_CHILD)).attributes.get(
-                ATTR_ID
-            ),
-            ATTR_START: call.data.get(ATTR_START),
-            ATTR_END: call.data.get(ATTR_END),
-            ATTR_NOTES: call.data.get(ATTR_NOTES),
+            ATTR_CHILD: self.child[ATTR_ID],
+            ATTR_TEMPERATURE: temperature,
+            ATTR_TIME: date_time,
         }
-
-        baby_buddy.entities_add(endpoint, data)
-
-    def services_temperature_add(call):
-        endpoint = ATTR_TEMPERATURE
+        if notes:
+            data[ATTR_NOTES] = notes
+
+        await self.coordinator.client.async_post(ATTR_TEMPERATURE, data)
+        await self.coordinator.async_request_refresh()
+
+    async def async_add_weight(
+        self, weight: float, date: date | None = None, notes: str | None = None
+    ) -> None:
+        """Add weight entry."""
+        if not isinstance(self, BabyBuddyChildSensor):
+            _LOGGER.debug("Babybuddy child sensor should be selected. Ignoring.")
+            return
         data = {
-            ATTR_CHILD: hass.states.get(call.data.get(ATTR_CHILD)).attributes.get(
-                ATTR_ID
-            ),
-            ATTR_TEMPERATURE: call.data.get(ATTR_TEMPERATURE),
-            ATTR_TIME: call.data.get(ATTR_TIME),
-            ATTR_NOTES: call.data.get(ATTR_NOTES),
+            ATTR_CHILD: self.child[ATTR_ID],
+            ATTR_WEIGHT: weight,
+            ATTR_DATE: date or dt_util.now().date(),
         }
+        if notes:
+            data[ATTR_NOTES] = notes
+
+        await self.coordinator.client.async_post(ATTR_WEIGHT, data)
+        await self.coordinator.async_request_refresh()
+
+    async def async_add_note(
+        self, note: str, time: datetime | time | None = None
+    ) -> None:
+        """Add note entry."""
+        if not isinstance(self, BabyBuddyChildSensor):
+            _LOGGER.debug("Babybuddy child sensor should be selected. Ignoring.")
+            return
+        try:
+            date_time = get_datetime_from_time(time or dt_util.now())
+        except ValidationError as err:
+            _LOGGER.error(err)
+            return
+        data = {ATTR_CHILD: self.child[ATTR_ID], ATTR_NOTE: note, ATTR_TIME: date_time}
 
-        baby_buddy.entities_add(endpoint, data)
+        await self.coordinator.client.async_post(ATTR_NOTES, data)
+        await self.coordinator.async_request_refresh()
 
-    """
-    def services_timers_add(call):
-        endpoint = ATTR_TIMERS
-        data = {}
+    async def async_delete_last_entry(self) -> None:
+        """Delete last data entry."""
+        if not isinstance(self, BabyBuddyChildDataSensor):
+            _LOGGER.debug("Babybuddy child data sensor should be selected. Ignoring.")
+            return
 
-        baby_buddy.entities_add(endpoint, data)
-    """
+        if self.state == None:
+            _LOGGER.error(f"{self.entity_description.key} entry is not available.")
+            return
```

Comment:
this won't work for entities like feedings where we are using a non-required field as the `state_key`.

maybe use `self.extra_state_attributes[ATTR_ID]`? should always exist if entry exists.

Reply:
I used the feeding amount as the state of the sensor so that the user can visualize the feeding amount during out the day when he clicks on the sensor to get the more-info card. Since the amount field is not required in babybuddy we should either default it to zero (otherwise it will show as `UNKNOWN` on the front end) or we can make it required in HA (with a  default value of zero in case the user doesn't wish to enter a value). What do you think?

Reply:
I prefer to leave as UNKNOWN, as that's more accurate. We just need to check for an attribute that will always exist in this `async_delete_last_entry `check.

Reply:
Done